include/circt/Dialect/FIRRTL/FIRRTLOpInterfaces.td

```diff
@@ -315,23 +315,31 @@ def InnerSymbol : OpInterface<"InnerSymbolOpInterface"> {
     InterfaceMethod<"Returns the name of this inner symbol.",
       "StringAttr", "getInnerNameAttr", (ins), [{}],
       /*defaultImplementation=*/[{
-        return this->getOperation()->template getAttrOfType<StringAttr>(
-            circt::firrtl::InnerSymbolTable::getInnerSymbolAttrName());
+        return InnerSymbolTable::getInnerSymbol(this->getOperation());
```

Comment:
One more thought, what if we don't make this change. Can we instead use `getInnerNameAttr` everywhere ? And ensure that only this `InterfaceMethod` can manually access the `sym_name` attribute. Is there any benefit or use of having `InnerSymbolTable::getInnerSymbol` and `getInnerSymName` ([`FIRRTLOps.h`](https://github.com/llvm/circt/blob/main/include/circt/Dialect/FIRRTL/FIRRTLOps.h#L99))